### PR TITLE
Add NusaX Bitext Mining (MINERS)

### PR DESCRIPTION
**Pull request by gentaiscool**
NusaX is a parallel dataset for machine translation and sentiment analysis on 11 Indonesian languages

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: ... 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            "revision": "fba4f2cfe2592641056f7a274c9aa8453b27a4a8",
+        },
+        description="NusaX is a parallel dataset for machine translation and sentiment analysis on 11 Indonesia languages and English.",
+        reference="https://huggingface.co/datasets/indonlp/NusaX-senti/",
+        type="BitextMining",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=_LANGUAGES,
+        main_score="f1",
+        date=("2023-01-01", "2023-03-01"),
```
Comment (KennethEnevoldsen) [inline mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:36]: This is intended to be data creation dates (when it was written). Can be an estimate

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            title={MINERS: Multilingual Language Models as Semantic Retrievers}, 
+            author={Genta Indra Winata and Ruochen Zhang and David Ifeoluwa Adelani},
+            year={2024},
+            eprint={2406.07424},
+            archivePrefix={arXiv},
+            primaryClass={cs.CL}
+        }
+        """,
+        n_samples={"train": 500},
+        avg_character_length={},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:63]: required

**Review by KennethEnevoldsen — COMMENTED** [review]
Only a few minor comments. For points let us just use the point system for adding datasets. It is 2 points for the dataset + 4 pr. new language task pair. 2 points are also given for a review. We can either do it per. PR or at the end.

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py (hunk 1 above)
Comment (gentaiscool) [reply mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py]: I just updated the date range to `date=("2021-08-01", "2022-07-01")`

Comment (gentaiscool) [reply mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py]: That was the time when we collected the dataset

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py (hunk 2 above)
Comment (gentaiscool) [reply mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py]: I added the char length.

**Comment by gentaiscool** [comment]
Thank you for the review. I have addressed your suggestion. @KennethEnevoldsen 

I found two issues:
1. when running `make test`
```
===================================================== short test summary info =====================================================
FAILED tests/test_RerankingEvaluator.py::TestRerankingEvaluator::test_nAUC - assert 0.9406528189910982 == 0.86943 ± 8.7e-05
```
It seemed unrelated to my change. I also got the error when running the command from the `main` branch.

2. And, I could not run `intfloat/multilingual-e5-small` model.
```
ValueError: Model None not found for model intfloat/multilingual-e5-small
```

**Comment by gentaiscool** [comment]
I have completed the checklist.

I have addressed your comments @KennethEnevoldsen. I could not run `intfloat/multilingual-e5-base` since I got this error `ValueError: Model None not found for model intfloat/multilingual-e5-small` so as a replacement I evaluated the dataset on `sentence-transformers/LaBSE`.

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            title={MINERS: Multilingual Language Models as Semantic Retrievers}, 
+            author={Genta Indra Winata and Ruochen Zhang and David Ifeoluwa Adelani},
+            year={2024},
+            eprint={2406.07424},
+            archivePrefix={arXiv},
+            primaryClass={cs.CL}
+        }
+        """,
+        n_samples={"train": 5500},
+        avg_character_length={"train": 5.00},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:63]: Is it on average 5 characters?

**Comment by KennethEnevoldsen** [comment]
> I could not run intfloat/multilingual-e5-base since I got this error ValueError: Model None not found for model intfloat/multilingual-e5-small so as a replacement I evaluated the dataset on sentence-transformers/LaBSE.

@gentaiscool will you add the results to the PR as well. 

> ValueError: Model None not found for model intfloat/multilingual-e5-small'

Addressed in #888 so merging from main should fix this

Will you add the points as well?

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py (hunk 3 above)
Comment (gentaiscool) [reply mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:63]: Sorry I put the number of words. Let me recalculate the number.

**Comment by gentaiscool** [comment]
thank you @KennethEnevoldsen for the review and now the evaluation on e5 model works :). I have updated the metadata (avg num of characters -- previously I put the avg of char per word, now should be correct) and add the model results.

File: mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py]: No worries. My guess was that it was something like that :)

**Comment by KennethEnevoldsen** [comment]
Points sounds reasonable. set tests to run and enable auto-merge if they pass

**Pull request merged**